### PR TITLE
Adding tests for Store API

### PR DESCRIPTION
**Pull request by nu1silva**
### Proposed changes in this pull request

[List all changes you want to add here. If you fixed an issue, please
add a reference to that issue as well.]

-

### When should this PR be merged

Depends on https://github.com/wso2/carbon-apimgt/pull/4354

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -827,7 +840,8 @@
         <hikari.package.import.version.range>[2.4.1, 3.0.0)</hikari.package.import.version.range>
         <javax.ws.rs.import.version.range>[2.0.0,3.0.0)</javax.ws.rs.import.version.range>
         <apache.commons.lang3.import.version.range>[3.4,4)</apache.commons.lang3.import.version.range>
-        <org.wso2.carbon.security.caas.import.version.range>[1.0.0,2.0.0)</org.wso2.carbon.security.caas.import.version.range>
+        <org.wso2.carbon.security.caas.import.version.range>[1.0.0,2.0.0)
```
Comment (harsha89) [inline pom.xml]: Can we have this in single line

File: pom.xml [hunk 2]
```diff
@@ -776,11 +785,15 @@
...
-        <osgi.framework.javax.xml.bind.package.import.version.range>[0.0.0, 1.0.0)</osgi.framework.javax.xml.bind.package.import.version.range>
+        <osgi.framework.console.package.import.version.range>[1.1.0, 2.0.0)
+        </osgi.framework.console.package.import.version.range>
+        <osgi.framework.javax.xml.bind.package.import.version.range>[0.0.0, 1.0.0)
+        </osgi.framework.javax.xml.bind.package.import.version.range>
         <osgi.services.cm.package.import.version.range>[1.5.0, 2.0.0)</osgi.services.cm.package.import.version.range>
-        <equinox.osgi.services.package.import.version.range>[1.0.0, 2.0.0)</equinox.osgi.services.package.import.version.range>
-        <osgi.service.tracker.package.import.version.range>[1.5.0, 2.0.0)</osgi.service.tracker.package.import.version.range>
+        <equinox.osgi.services.package.import.version.range>[1.0.0, 2.0.0)
+        </equinox.osgi.services.package.import.version.range>
```
Comment (harsha89) [inline pom.xml]: Can we have this in single line

File: pom.xml [hunk 3]
```diff
@@ -776,11 +785,15 @@
         <osgi.framework.import.version.range>[1.8.0, 2.0.0)</osgi.framework.import.version.range>
         <osgi.service.tracker.import.version.range>[1.5.1, 2.0.0)</osgi.service.tracker.import.version.range>
         <osgi.framework.package.import.version.range>[1.8.0, 2.0.0)</osgi.framework.package.import.version.range>
-        <osgi.framework.console.package.import.version.range>[1.1.0, 2.0.0)</osgi.framework.console.package.import.version.range>
-        <osgi.framework.javax.xml.bind.package.import.version.range>[0.0.0, 1.0.0)</osgi.framework.javax.xml.bind.package.import.version.range>
+        <osgi.framework.console.package.import.version.range>[1.1.0, 2.0.0)
```
Comment (harsha89) [inline pom.xml]: Can we have this in single line

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/pom.xml [hunk 4]
```diff
@@ -68,8 +69,18 @@
             <artifactId>jackson-jaxrs-json-provider</artifactId>
         </dependency>
         <dependency>
-            <groupId>org.testng</groupId>
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/pom.xml]: You can keep the testng dependency and used when it required

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/test/java/org/wso2/carbon/apimgt/rest/api/store/mappings/CommentMappingUtilTestCase.java [hunk 5]
```diff
@@ -19,8 +19,8 @@
  */
 package org.wso2.carbon.apimgt.rest.api.store.mappings;
 
-import org.testng.Assert;
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/test/java/org/wso2/carbon/apimgt/rest/api/store/mappings/CommentMappingUtilTestCase.java]: No need to change it if u hv the testng dependency

Comment (nu1silva) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store/src/test/java/org/wso2/carbon/apimgt/rest/api/store/mappings/CommentMappingUtilTestCase.java]: tests already converted to junit. Lets keep them as it is for now. TestNG dependency was added back so we can write testng tests aswell.

**Pull request merged**